### PR TITLE
[14.0][FIX]sale_commission: settlement report fixes

### PR DESCRIPTION
**Pull request by PicchiSeba**
EDIT:

This PR contains the following changes to the settlement report

- fixes to the spacing of certain elements
~~removed the field "Invoice" because it is always empty~~

### CODE REVIEW

File: sale_commission/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    show_partner_settlement_report = fields.Boolean(
+        "Show partner details in settlements report",
+        default=False,
+    )
+    settlement_skip_ero_amount_lines = fields.Boolean(
```
Comment (francesco-ooops) [inline sale_commission/models/res_company.py]: typo (also elsewhere)

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

in this PR:

- fixed settlement report ("Invoice" field was empty but was not relevant as the same info are present in "Invoice line" field)
- Added option to display partner in settlement lines report
- Added option to not display settlement lines with zero amount (useful with _product_criteria modules which generate many such lines)

File: sale_commission/readme/CONFIGURE.rst [hunk 2]
```diff
@@ -48,3 +48,17 @@ For adding new agents:
...
+#. Go to *Settings > Sales > Quotations & Orders*:
+
+  * enable he setting "Show partner details in settlements report"
+
+
+To hide lines with amount equals to 0:
+
+#. Go to *Settings > Sales > Quotations & Orders*:
+
+  * enable he setting "Skip lines in settlements with zero amount"
```
Comment (renda-dev) [inline sale_commission/readme/CONFIGURE.rst]: **typo:**

```suggestion
  * enable the setting "Show partner details in settlements report"


To hide lines with amount equals to 0:

#. Go to *Settings > Sales > Quotations & Orders*:

  * enable the setting "Skip lines in settlements with zero amount"
```

**Review by renda-dev — APPROVED** [review]
Code review, LGTM

**Comment by francesco-ooops** [comment]
@pedrobaeza what do you say? :)

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
For me this is overcomplicating the module introducing computed fields that slows down all the usual workflows. This should be IMO in a separated module for those wanting that specific behavior.

Apart from that, computed one2many are not officially supported, and can give some glitches. They should be m2m.

**Comment by francesco-ooops** [comment]
@pedrobaeza shall we at least leave the commit where we fix report?

**Comment by pedrobaeza** [comment]
OK.

**Comment by PicchiSeba** [comment]
Ok @pedrobaeza.

I removed all the unnecessary changes from this PR (I will eventually create a new module in a new PR). I only left the fixes regarding spacing in the report

**Comment by francesco-ooops** [comment]
@PicchiSeba please reinstate the fix with the removal of column "invoice" from report

**Comment by PicchiSeba** [comment]
I also updated the PR's title and description

**Comment by pedrobaeza** [comment]
Now reading the restricted fix, this is not really true. The `invoice_id` field is filled with the vendor bill that is done for outsiders. Your DB will be empty if all the commission agents are internal, but we shouldn't remove it.

**Comment by PicchiSeba** [comment]
That's exactly our case. Good to know. I'm removing the offending commit

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Comment by francesco-ooops** [comment]
> Now reading the restricted fix, this is not really true. The `invoice_id` field is filled with the vendor bill that is done for outsiders. Your DB will be empty if all the commission agents are internal, but we shouldn't remove it.

@pedrobaeza I think our use case is a settlement report where agent vendor bill hasn't been created yet, so that condition could exclude the field from the report. We'll think of such change

**Comment by pedrobaeza** [comment]
Maybe you can detect if there's any invoice in whole report, and it not, hide the column.

**Pull request merged**